src/UtilMsg.chpl

```diff
@@ -21,127 +23,81 @@ module UtilMsg {
 
     see: https://numpy.org/doc/stable/reference/generated/numpy.diff.html
   */
-  @arkouda.registerND
-  proc diffMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, param nd: int): MsgTuple throws {
-    param pn = Reflection.getRoutineName();
-
-    const name = msgArgs.getValueOf("name"),
-          n = msgArgs.get("n").getIntValue(),
-          axis = msgArgs.get("axis").getPositiveIntValue(nd),
-          rname = st.nextName();
-
-    var gEnt: borrowed GenSymEntry = getGenericTypedArrayEntry(name, st);
-
-    proc doDiff(type t): MsgTuple throws {
-      const e = toSymEntry(gEnt, t, nd);
-
-      if n == 1 {
-        // 1st order difference: requires no temporary storage
-        const outDom = subDomain(e.tupShape, axis, 1);
-        var d = st.addEntry(rname, (...outDom.shape), t);
-
-        forall axisSliceIdx in domOffAxis(e.a.domain, axis) {
-          const slice = domOnAxis(outDom, tuplify(axisSliceIdx), axis);
-          for i in slice {
-            var idxp = tuplify(i);
-            idxp[axis] += 1;
-            d.a[i] = e.a[idxp] - e.a[i];
-          }
+
+  @arkouda.registerCommand()
+  proc diff(x: [?d] ?t, n: int, axis: int): [] t throws 
+    where (t == real) || (t == int) || (t == uint(8)) || (t == uint(64)){
+
+    const outDom = subDomain(x.shape, axis, n);
+    if n == 1 {
+      // 1st order difference: requires no temporary storage
+      var y = makeDistArray(outDom, t);
+      for axisSliceIdx in domOffAxis(d, axis) {
+        const slice = domOnAxis(outDom, tuplify(axisSliceIdx), axis);
+        for i in slice {
+          var idxp = tuplify(i);
+          idxp[axis] += 1;
+          y[i] = x[idxp] - x[i];
         }
-      } else {
-        // n'th order difference: requires 2 temporary arrays
-        var d1 = makeDistArray(e.a);
-
-        {
-          var d2 = makeDistArray(e.a.domain, e.a.eltType);
-          for m in 1..n {
-            d1 <=> d2;
-            const diffSubDom = subDomain(e.tupShape, axis, m);
-
-            forall axisSliceIdx in domOffAxis(e.a.domain, axis) {
-              const slice = domOnAxis(diffSubDom, tuplify(axisSliceIdx), axis);
-
-              for i in slice {
-                var idxp = tuplify(i);
-                idxp[axis] += 1;
-                d1[i] = d2[idxp] - d2[i];
-              }
+      }
+      return y;
+    } else {
+      // n'th order difference: requires 2 temporary arrays
+      var d1 = makeDistArray(x);
+      {
+        var d2 = makeDistArray(d, t);
+        for m in 1..n {
+          d1 <=> d2;
+          const diffSubDom = subDomain(x.shape, axis, m);
+
+          forall axisSliceIdx in domOffAxis(d, axis) {
+            const slice = domOnAxis(diffSubDom, tuplify(axisSliceIdx), axis);
+
+            for i in slice {
+              var idxp = tuplify(i);
+              idxp[axis] += 1;
+              d1[i] = d2[idxp] - d2[i];
             }
           }
-        } // d2 deinit here
```

Comment:
could you keep this comment (it explains why d2's declaration needs to be in a nested scope).

Reply:
I put it back!